packages/astro/src/i18n/index.ts

```diff
@@ -244,3 +276,109 @@ class Unreachable extends Error {
 		);
 	}
 }
+
+export type MiddlewarePayload = {
+	base: string;
+	locales: Locales;
+	trailingSlash: AstroConfig['trailingSlash'];
+	format: AstroConfig['build']['format'];
+	routing?: RoutingStrategies;
+	defaultLocale: string;
+	domains: Record<string, string> | undefined;
+	fallback: Record<string, string> | undefined;
+};
+
+export function redirectToDefaultLocale({
+	trailingSlash,
+	format,
+	base,
+	defaultLocale,
+}: MiddlewarePayload) {
+	return function (context: APIContext, statusCode?: ValidRedirectStatus) {
+		if (shouldAppendForwardSlash(trailingSlash, format)) {
+			return context.redirect(`${appendForwardSlash(joinPaths(base, defaultLocale))}`, statusCode);
+		} else {
+			return context.redirect(`${joinPaths(base, defaultLocale)}`, statusCode);
+		}
+	};
+}
+
+export function notFound({ base, locales }: MiddlewarePayload) {
```

Comment:
are these new exported functions intended to be used as helpers?

Reply:
Yes, as per description:
> I refactored our internal middleware. I took shared functions and lifted them, so they can be used inside our internal middleware and the user middleware

I thought it was a good way to dogfooding the helpers that we expose to the users, because we use them too.

Reply:
Might be useful to add a comment to say those functions are public to users

Reply:
Done in [`e75b46c` (#10193)](https://github.com/withastro/astro/pull/10193/commits/e75b46c01fe5a3284a59e91c15a2cb79cd5dfc6b)